src/UserActivityFeedEventSchemas/AcademicProfileTagCreatedEventSchema.js

```diff
@@ -0,0 +1,34 @@
+module.exports = (conn) => {
+  const AcademicProfileTagCreatedEventSchema = new conn.Schema({
+    createdBy: {
+      type: conn.Schema.Types.ObjectId,
+      ref: 'User',
+      required: true,
+    },
+    tag: {
+      type: String,
+      required: true,
+    },
+    assignmentReason: {
+      type: String,
+      required: true,
+      trim: true,
+    },
+  });
+
+  // NOTE: The tests of this callback are the repository of models
+  AcademicProfileTagCreatedEventSchema.pre('save', function (next) {
```

Comment:
I believe We are creating mongoose hooks in `models` repo rather than `schemas` repo.

can we move it to  `models` repo.